401d56/code-challenges/gradle/src/main/java/hashtable/Hashtable.java

```diff
@@ -0,0 +1,56 @@
+package hashtable;
+
+public class Hashtable {
+
+    private Node[] map;
+
+    public Hashtable(int size){
+        map = new Node[size];
+    }
+
+    int hash(String key){
+        int hashValue = 0;
+        char[] letters = key.toCharArray();
+        for(int i = 0; i < letters.length; i++){
+            hashValue += letters[i];
+        }
+        hashValue = (hashValue * 843) % map.length;
+        return hashValue;
+    }
+
+    public void set(String key, String value) {
+        int hashkey = hash(key);
+
+        if (this.map[hashkey] == null) {
+            map[hashkey] = new Node(key, value);
+        } else {
```

Comment:
What should be done if a key already exists in the map? You add it to the list, but it can't ever be found, you return the first node with the key you're looking for